sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java

```diff
@@ -48,18 +48,20 @@ public String getApplicationId() {
      * {@codesnippet com.azure.core.util.ClientOptions.setApplicationId#String}
      *
      * @param applicationId The application ID.
+     *
      * @return The updated ClientOptions object.
+     *
      * @throws IllegalArgumentException If {@code applicationId} contains spaces or larger than 24 in length.
      */
     public ClientOptions setApplicationId(String applicationId) {
-        if (CoreUtils.isNullOrEmpty(applicationId)) {
-            this.applicationId = applicationId;
-        } else if (applicationId.length() > MAX_APPLICATION_ID_LENGTH) {
-            throw logger.logExceptionAsError(new IllegalArgumentException(INVALID_APPLICATION_ID_LENGTH));
-        } else if (applicationId.contains(" ")) {
-            throw logger.logExceptionAsError(new IllegalArgumentException(INVALID_APPLICATION_ID_SPACE));
-        } else {
-            this.applicationId = applicationId;
+        if (!CoreUtils.isNullOrEmpty(applicationId)) {
+            if (applicationId.length() > MAX_APPLICATION_ID_LENGTH) {
+                throw logger.logExceptionAsError(new IllegalArgumentException(INVALID_APPLICATION_ID_LENGTH));
+            } else if (applicationId.contains(" ")) {
+                throw logger.logExceptionAsError(new IllegalArgumentException(INVALID_APPLICATION_ID_SPACE));
+            } else {
+                this.applicationId = applicationId;
```

Comment:
I'd still set `applicationId` if the passed value is null. It'll allow for resetting of the `applicationId`.